parsons/databases/db_sync.py

```diff
@@ -0,0 +1,182 @@
+
+    def table_sync_incremental(self, source_table, destination_table, primary_key,
+                               distinct_check=True, **kwargs):
+        """
+        Incremental sync of table from a source database to a destination database
+        using an incremental primary key.
+
+        `Args:`
+            source_table: str
+                Full table path (e.g. ``my_schema.my_table``)
+            destination_table: str
+                Full table path (e.g. ``my_schema.my_table``)
+            if_exists: str
+                If destination table exists either ``drop`` or ``truncate``. Truncate is
+                useful when there are dependent views associated with the table.
+            primary_key: str
+                The name of the primary key. This must be the same for the source and
+                destination table.
+            distinct_check: bool
+                Check that the source table primary key is distinct prior to running the
+                sync. If it is not, an error will be raised.
+            **kwargs: args
+                Optional copy arguments for destination database.
+        `Returns:`
+            ``None``
+        """
+
+        # Create the table objects
+        source_tbl = self.source_db.table(source_table)
+        destination_tbl = self.dest_db.table(destination_table)
+
+        # Check that the destination table exists. If it does not, then run a
+        # full sync instead.
+        if not destination_tbl.exists:
+            self.table_sync_full(source_table, destination_table)
+
+        # Check that the source table primary key is distinct
+        if distinct_check and not source_tbl.distinct_primary_key(primary_key):
+            raise ValueError('{primary_key} is not distinct in source table.')
+
+        # Get the max source table and destination table primary key
+        source_max_pk = source_tbl.max_primary_key(primary_key)
+        dest_max_pk = destination_tbl.max_primary_key(primary_key)
+
+        # Check for a mismatch in row counts.
+        if dest_max_pk > source_max_pk:
+            raise ValueError('Destination DB primary key greater than source DB primary key.')
+
+        # Do not copied if row counts are equal.
+        elif dest_max_pk == source_max_pk:
+            logger.info('Tables are in sync.')
+            return None
+            # Question: Should I check that the row counts match too?
+
+        else:
+            # Get count of rows to be copied.
+            new_row_count = source_tbl.get_new_rows_count(primary_key, dest_max_pk)
+            logger.info(f'Found {new_row_count} new rows in source table.')
+
+            # Copy rows in chunks.
+            copied_rows = 0
+            while copied_rows < new_row_count:
+
+                # Get a chunk
+                rows = source_tbl.get_new_rows(primary_key=primary_key,
+                                               max_value=dest_max_pk,
```

Comment:
I think `max_value` is a bit of a misnomer. Maybe `start_value` would work as a name, since it's the value you are starting from as you paginate through the table?